### DISCUSSION TITLE
[IOInterface] the methods $context parameter

### DISCUSSION THREAD

**Discussion by rino1** [composer/composer#11882] [General]
https://getcomposer.org/doc/articles/plugins.md

Let's summarize like this

````
    public function activate(Composer $composer, IOInterface $io)
    {
        printf(
            "Debug: %d\n",
            $io->isDebug()
        );
        $io->debug(
            message: "ABC",
        );
        $io->debug(
            message: "ABC",
            context: ['XYZ']
        );
````

We summarize the output produced

````
Debug: 1
ABC
ABC
````

Note that whether the context parameter (context: ['XYZ']) is specified or not, the output is always the same (ABC).

So what is this parameter for?

**Comment by Seldaek** [comment]
The context is just present for the LoggerInterface, but it is not used/output by composer's implementation of the log interface as we do not use it anywhere..

Anyway it is fixed by https://github.com/composer/composer/commit/2124f09d75167449e9efa8ae3e2d8ad034e64bdb - please try again after a `composer self-update --snapshot`

**Comment by rino1** [comment]
$ composer self-update --snapshot

                                         
  Command "self-update" is not defined.

**Reply by Seldaek** [reply]
Then install composer from the official installer and not system packages if you want to try the latest.

**Comment by rino1** [comment]
````
$ sudo apt install composer
Lettura elenco dei pacchetti... Fatto
Generazione albero delle dipendenze... Fatto
Lettura informazioni sullo stato... Fatto   
composer è già alla versione più recente (2.2.6-2ubuntu4).
0 aggiornati, 0 installati, 0 da rimuovere e 4 non aggiornati.
~$ 
````

**Reply by Seldaek** [reply]
I mean this is the official installer https://getcomposer.org/download/

**Comment by rino1** [comment]
thanks
